src/app/app.component.ts

```diff
@@ -12,14 +12,103 @@ import { Component, OnInit } from '@angular/core';
 })
 export class AppComponent implements OnInit {
   readonly DEFAULT_BANNER_DESCRIPTION: string = "Safer, faster, and more stable"
+  readonly ANDROID_CHROME_CLICK_DESCRIPTION = "Click to open link with Banana browser"
+
+  readonly PACKAGE_ID = "org.triple.banana"
+  readonly DEFAULT_WAIT_TIME_MS = 300
+
+  readonly TRIPLE_BANANA_DEV_URL = "https://triplebanana.dev"
+  readonly PLAYSTORE_URL = "https://play.google.com/store/apps/details"
+
   bannerDescription = this.DEFAULT_BANNER_DESCRIPTION
 
+  private targetUrl: URL = new URL(this.TRIPLE_BANANA_DEV_URL)
+  private targetUrlOnly: string = ""
+
   ngOnInit(): void {
-    // Parse parameters as url and move
+    this.processShareLink()
   }
 
   onBannerClicked() {
-    // If browser is chromium based,
-    // it is allowed to move only when user click action
+    this.gotoLinkOrStore(this.targetUrlOnly)
+  }
+
+  private processShareLink() {
+    const linkTo = this.getLinkToUrl()
+
+    if (!linkTo) {
+      console.error("Failed to parse banana link : " + window.location.toString())
+      this.moveToUrl(this.TRIPLE_BANANA_DEV_URL)
+      return
+    }
+    this.targetUrl = linkTo
+
+    this.targetUrlOnly = linkTo.toString().substring(linkTo.protocol.length + "//".length)
+    this.gotoLinkOrStore(this.targetUrlOnly)
+  }
+
+  private gotoLinkOrStore(url: string) {
+    const userAgent = this.getUserAgent();
+    if (userAgent.match(/Android/)) {
+      if (userAgent.match(/Chrome/)) {
+        this.handleChromium(url)
+      } else {
+        this.handleOtherBrowsers(url)
+      }
+    }
+    else {
+      this.moveToUrl(this.targetUrl.toString())
```

Comment:
If we use `TRIPLE_BANANA_DEV_URL` as a parameter,  `targetUrl` seems like it can be removed.